File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        // set<int> n ;
        // for(int x : nums){
        //     if(n.find(x) != n.end())return true ;
        //     n.insert(x);
        // }
        // return false;
        int n = (int)nums.size();
        sort(nums.begin(),nums.end());
        for(int i = 0 ; i < n-1 ; ++i)if(nums[i]==nums[i+1])return true;
        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.ex
defmodule Solution do
  @spec contains_duplicate(nums :: [integer]) :: boolean
  def contains_duplicate(nums) do
    Enum.reduce_while(nums, MapSet.new(), fn x, seen ->
      if MapSet.member?(seen, x) do
        {:halt, true}
      else
        {:cont, MapSet.put(seen, x)}
      end
    end) == true
  end
end
